packages/@react-spectrum/sidenav/test/SideNav.test.js

```diff
@@ -102,4 +136,220 @@ describe('SideNav', function () {
+  `('$Name can render sections', async function ({Name, Component, ComponentSection, ComponentItem}) {
+    let {getByText, getAllByRole} = renderComponent(Name, Component, ComponentSection, ComponentItem);
+
+    await waitForDomChange();
+
+    let [section1, section2] = [
+      getByText('Section 1'),
+      getByText('Section 2')
+    ];
+
+    expect(section1).toBeTruthy();
+    expect(section2).toBeTruthy();
+
+    // contains 4 link items
+    expect(getAllByRole('link').length).toBe(4);
+    // all 4 items and 2 sections have role presentation
+    expect(getAllByRole('presentation').length).toBe(6);
+  });
+
+  it.each`
+    Name                           | Component    | ComponentSection | ComponentItem
+    ${'SideNav'}                   | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStatic'}             | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavWithSections'}       | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStaticWithSections'} | ${SideNav}   | ${Section}       | ${Item}
+  `('$Name can focus up/down item', async function ({Name, Component, ComponentSection, ComponentItem}) {
+    let {getAllByRole} = renderComponent(Name, Component, ComponentSection, ComponentItem);
+    
+    await waitForDomChange();
+
+    let items = getAllByRole('link');
+    let selectedItem = items[0];
+    selectedItem.focus();
+    expect(selectedItem).toBe(document.activeElement);
+    fireEvent.keyDown(selectedItem, {key: 'ArrowDown', code: 40, charCode: 40});
+    let nextSelectedItem = items[1];
+    expect(nextSelectedItem).toBe(document.activeElement);
+    fireEvent.keyDown(nextSelectedItem, {key: 'ArrowUp', code: 38, charCode: 38});
+    expect(selectedItem).toBe(document.activeElement);
+  });
+
+  it.each`
+    Name                           | Component    | ComponentSection | ComponentItem
+    ${'SideNav'}                   | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStatic'}             | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavWithSections'}       | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStaticWithSections'} | ${SideNav}   | ${Section}       | ${Item}
+  `('$Name can focus first to last/last to first item', async function ({Name, Component, ComponentSection, ComponentItem}) {
+    let {getAllByRole} = renderComponent(Name, Component, ComponentSection, ComponentItem);
+    
+    await waitForDomChange();
+
+    let items = getAllByRole('link');
+    let selectedItem = items[0];
+    selectedItem.focus();
+    expect(selectedItem).toBe(document.activeElement);
+    fireEvent.keyDown(selectedItem, {key: 'ArrowUp', code: 40, charCode: 40});
+    let nextSelectedItem = items[items.length - 1];
+    expect(nextSelectedItem).toBe(document.activeElement);
+    fireEvent.keyDown(nextSelectedItem, {key: 'ArrowDown', code: 38, charCode: 38});
+    expect(selectedItem).toBe(document.activeElement);
+  });
+
+  // TODO: add test for static collection
```

Comment:
Is this going to happen in this pull or a follow up one?
Everything in this pull so far lgtm

Reply:
added here